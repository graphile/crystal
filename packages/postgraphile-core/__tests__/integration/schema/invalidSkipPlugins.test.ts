const { withPgClient } = require("../../helpers");
const { createPostGraphileSchema } = require("../../..");

const PluginThatWouldNotHaveLoadedAnyway = () => {};

test("throws error when skipPlugins is passed a plugin that would not have loaded", () =>
  withPgClient(async (client) => {
    let err;
    try {
      await createPostGraphileSchema(client, "a", {
        skipPlugins: [PluginThatWouldNotHaveLoadedAnyway],
      });
    } catch (_e) {
      err = _e;
    }
    expect(err).toBeTruthy();
    expect(err).toMatchSnapshot();
  }));
