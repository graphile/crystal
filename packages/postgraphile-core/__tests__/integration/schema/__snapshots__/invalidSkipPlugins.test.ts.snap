// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`throws error when skipPlugins is passed a plugin that would not have loaded 1`] = `[Error: You tried to skip plugins that would never have been loaded anyway. Perhaps you've made a mistake in your skipPlugins list, or have sourced the plugin from a duplicate plugin module - check for duplicate modules in your 'node_modules' folder. The plugins that you requested to skip were: PluginThatWouldNotHaveLoadedAnyway]`;
